### DISCUSSION TITLE
Can I fine tuning a llama model with pcw?

### DISCUSSION THREAD

**Discussion by naknak-choi** [AI21Labs/Parallel-Context-Windows#8] [Q&A]
Hi there! (if it's rude let me know!)

In the paper, "we believe that further training of an LLM with pcw could further enhance the abilities demonstrated in this work"
So, if I just add training code, LlamaForCausalLMPCW model can be trained??

thank you!

**Comment by nirrai21** [comment]
Hi,
The short answer is yes, though though I suppose you will need to modify the default transformer's training.

**Reply by naknak-choi** [reply]
Thanks for replying!

That's mean when I give training data to transformer with pcw, I should override LlamaForCausalLM forward on LlamaForCausalLMPCW according to pcw attention?

**Reply by naknak-choi** [reply]
Then if I don't want use Hugging face Trainer means implement a training code, I don't need to modify the default transformer's training right???

**Reply by nirrai21** [reply]
I am afraid I won't be able to assist you with the technical issues here since we didn't implement any training mechanism in our paper. Good luck.
